.github/workflows/snyk.yml

```diff
@@ -0,0 +1,31 @@
+# This action runs every day at 6 AM and on every push
+# If the branch it's running on is main then it will run snyk monitor (reports vulnerabilities to snyk.io)
+# Otherwise it will run snyk test
+name: Snyk
+
+on:
+  schedule:
+    - cron: "0 6 * * *"
+  push:
```

Comment:
We're only interested in pushes to `main` branch so this could be restricted to that branch